### DISCUSSION TITLE
Add a variable to set deployment mode on complete or incremental?

### DISCUSSION THREAD

**Discussion by fabianjogschies** [Azure/data-management-zone#119] [Ideas] [answered]
As far as i see deployment mode is set to complete in order you got a failed deployment at some point it would be useful in my opinion to choose between incremental and complete.

**Comment by marvinbuss** [comment, marked as answer]
@fabianjogschies No we are not using `Complete` mode. We are using `Incremental` mode for all deployments. Otherwise you would run into other kinds of issues with Private Endpoints etc.
Do you require `Complete` mode? Is there a use case for that?

**Reply by fabianjogschies** [reply]
Ok my fault as is saw that the network resources took quiet long in the second deployment.
I think if you're starting with esa and maybe you're about to do a POC and test a lot it would be helpful to setup a clean environment. But this can also be done through the clean up command and than setup a new one.
Maybe this can be a difference between the different environments dev/test/prod?

**Reply by marvinbuss** [reply]
With complete mode, you will run into some issues with private endpoints and the corresponding NICs, which is why `Complete` mode is not recommended at this point in time. More can be found here: https://feedback.azure.com/forums/217313-networking/suggestions/40395946-private-endpoint-arm-template-deployment-fix-comp
You can find the same issue being reported elsewhere.

If we can close this, then please mark any comment as `Answer`. 👍 
Thanks!
